src/common/FrameManager.ts

```
@@ -317,24 +380,26 @@ export class FrameManager extends EventEmitter {
     this.emit(FrameManagerEmittedEvents.FrameNavigated, frame);
   }
 
-  async _ensureIsolatedWorld(name: string): Promise<void> {
-    if (this._isolatedWorlds.has(name)) return;
-    this._isolatedWorlds.add(name);
-    await this._client.send('Page.addScriptToEvaluateOnNewDocument', {
+  async _ensureIsolatedWorld(session: CDPSession, name: string): Promise<void> {
+    const key = `${session.id()}:${name}`;
+    if (this._isolatedWorlds.has(key)) return;
+    this._isolatedWorlds.add(key);
+
+    await session.send('Page.addScriptToEvaluateOnNewDocument', {
       source: `//# sourceURL=${EVALUATION_SCRIPT_URL}`,
       worldName: name,
     });
     // Frames might be removed before we send this.
     await Promise.all(
-      this.frames().map((frame) =>
-        this._client
-          .send('Page.createIsolatedWorld', {
+      this.frames()
+        .filter((frame) => frame._client === session)
+        .map((frame) =>
+          session.send('Page.createIsolatedWorld', {
             frameId: frame._id,
             worldName: name,
             grantUniveralAccess: true,
           })
-          .catch(debugError)
```

why don't we need debugError anymore?

---

Debug error is potentially hiding important errors. Instead, I chose to add a try/catch block in initialize that ignores the errors that should be ignored and throws all others.